vis/graphs.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-undef */
+
+var num_projects = [];
+var y_descriptions = [];
+var type_quarter = [];
+
+const number_project_types = 10;
+const number_quarters = 12;
+
+d3.json('projects-extended.json').then((res) => {
+    Object.keys(res).forEach(year => {
+        const quarters = Object.keys(res[year]).reverse();
+        const quartersinfo = Object.values(res[year]).reverse();
+        for(let i = 0; i < Object.values(quarters).length; i++){
+            let quarter = year + ' ' + Object.values(quarters)[i];
+            type_quarter.push(quarter);
+            num_projects.push(Object.keys(Object.values(quartersinfo)[i]).length-3);
+
+            let project = Object.values(quartersinfo)[i];
+            project_desc = Object.values(project).filter(x => x.Description);
+            y_descriptions.push(project_desc);
+        }
+    });
+
+    let proj_type_num = Array(number_project_types).fill(0);
+
+    let proj_type_quarter = Array(number_quarters).fill().map(() =>
+        Array(number_project_types).fill(0),
+    );
+
+    let quarter_count = 0;
```

Comment:
lets take advantage of the index increasing itself!

remove quarter_count and use the index property of the `Array.map` function